src/Transport.cpp

```diff
@@ -396,6 +452,32 @@ namespace mediasoupclient
 		return consumer;
 	}
 
+	/**
+	 * Create a DataConsumer.
+	 */
+	DataConsumer* RecvTransport::ConsumeData(DataConsumer::Listener* listener, 
+		DataConsumerOptions dataConsumerOptions)
+	{
+		MSC_TRACE();
+		if (this->closed)
+			MSC_THROW_INVALID_STATE_ERROR("RecvTransport closed");
+		else if (dataConsumerOptions.dataConsumerId.empty())
+			MSC_THROW_TYPE_ERROR("missing dataConsumerId");
+		else if (dataConsumerOptions.dataProducerId.empty())
+			MSC_THROW_TYPE_ERROR("missing producerId");
+		else if (!this->hasSctpParameters) 
+			MSC_THROW_TYPE_ERROR("Cannot use DataChannels with this transport. SctpParameters were not set for this transport.");
+		
+		RecvHandler::RecvDataChannel res = this->recvHandler->CreateRecvDataChannel(dataConsumerOptions.label, dataConsumerOptions);
```

Comment:
```suggestion
		// This may throw.
		RecvHandler::RecvDataChannel res = this->recvHandler->CreateRecvDataChannel(dataConsumerOptions.label, dataConsumerOptions);
```